src/main/java/dev/jbang/Util.java

```diff
@@ -531,36 +531,63 @@ public static String getStableID(String input) {
 	private static String extractFileFromGist(String url) {
 
 		try {
-			String gistapi = url.replaceFirst("^https://gist.github.com/(([a-zA-Z0-9]*)/)?(?<gistid>[a-zA-Z0-9]*)$",
+			String[] pathPlusAnchor = url.split("#");
+			String fileName = "";
+			if (pathPlusAnchor.length == 2) {
+				String[] anchor = pathPlusAnchor[1].split("-");
+				if (anchor.length < 2)
+					throw new IllegalArgumentException("Invalid Gist url: " + url);
+				fileName = anchor[1];
+				for (int i = 2; i < anchor.length - 1; ++i)
+					fileName += "-" + anchor[i];
+			}
+			String gistapi = pathPlusAnchor[0].replaceFirst(
+					"^https://gist.github.com/(([a-zA-Z0-9]*)/)?(?<gistid>[a-zA-Z0-9]*)$",
 					"https://api.github.com/gists/${gistid}");
-			// Util.info("looking at " + gistapi);
+
+			Util.verboseMsg("looking at " + gistapi);
 			String strdata = null;
 			try {
 				strdata = readStringFromURL(gistapi);
 			} catch (IOException e) {
-				// Util.info("error " + e);
+				Util.verboseMsg("error " + e);
 				return url;
 			}
 
 			Gson parser = new Gson();
 
 			Gist gist = parser.fromJson(strdata, Gist.class);
 
-			// Util.info("found " + gist.files);
-			final Optional<Map.Entry<String, Map<String, String>>> first = gist.files	.entrySet()
-																						.stream()
-																						.filter(e -> e	.getKey()
-																										.endsWith(
-																												".java")
-																								|| e.getKey()
-																									.endsWith(".jsh"))
-																						.findFirst();
-
-			if (first.isPresent()) {
-				// Util.info("looking at " + first);
-				return (String) first.get().getValue().getOrDefault("raw_url", url);
+			Util.verboseMsg("found " + gist.files);
```

Comment:
same as before, if putting verbose message in then add some context to be informative/useful.